classes/indoorFloor.tsx

```diff
@@ -39,59 +45,174 @@ class IndoorFloor {
-                    color = 'yellow'
-                }
-            })
+        else if (type === "waterfountain"){
+            return {
+                zIndex: 7,
+                position: 'absolute',
+                top: yPosition,
+                left: xPosition,
+                borderRadius: 23 / 2,
+                width: 23,
+                height: 23,
+                opacity: 1
+            }
         }
-        if(this.floorData.waterFountains) {
-            this.floorData.waterFountains.forEach(waterFountain =>{
-                if(xIndex === waterFountain.x && yIndex === waterFountain.y) {
-                    color = 'purple'
-                }
-            })
+        else if (type === "elevator"){
+            return {
+                zIndex: 7,
+                    position: 'absolute',
+                    top: yPosition,
+                    left: xPosition,
+                    width: 25,
+                    height: 25,
+                    opacity: 1
+            }
         }
-        if(this.floorData.classRooms) {
-            this.floorData.classRooms.forEach(classRoom =>{
-                if(xIndex === classRoom.location.x && yIndex === classRoom.location.y) {
-                    color = 'orange'
-                }
-            })
+        else if (type === "staircase"){
+            return {
+                zIndex: 7,
+                    position: 'absolute',
+                    top: yPosition,
+                    left: xPosition,
+                    width: 25,
+                    height: 25,
+                    opacity: 1
+            }
         }
-        return color;
+        return {}
     }
 
+    // this function is called when a set of coordina  i - z return a match with the TYPE of indoor poit
+    // this function returns an image (logo) of the specified indoor poi
+    // function is called in showIndoorTile()
+    // this function calls setLogoStyle to apply the correct style to each type of icon
+    setIndoorPOILogo(i, z, xPosition, yPosition, type) {
+        let img
+
+        // toilet icon
+        if (type === "toilet") {
+            img = <Image
+                source={toiletIcon}
+                key={i + " " + z}
+                style={this.setLogoStyle(xPosition, yPosition, type)}
+            >
+            </Image>
+        }
+        // waterfountain icon
+        else if (type === "waterfountain") {
+            img = <Image
+                source={waterFountainIcon}
+                key={i + " " + z}
+                style={this.setLogoStyle(xPosition, yPosition, type)}>
+            </Image>
+        }
+        // staircase icon
+        else if (type === "staircase") {
+            img = <Image
+                source={staircaseIcon}
+                key={i + " " + z}
+                style={this.setLogoStyle( xPosition, yPosition, type)}>
+            </Image>
+        }
+        // elevator icon
+        else if (type === "elevator") {
```

Comment:
As mentioned in my previous comment, you duplicate a lot of code. The only thing that changes in 
```
<Image
                source={staircaseIcon}
                key={i + " " + z}
                style={this.setLogoStyle( xPosition, yPosition, type)}>
            </Image>
```
is the image source. So instead of re-writing all the image tags, make a switch statement that will set the imageIcon, then all you will need is 1 Image tag